Cyberlimb expansion

Citrea-Lingua opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
General overhaul and expansion of cyberware.

## Why we need to add this
Cyberlimbs, right now, are great! But also a little... limited. The amount of times anyone has come to medical to ask for a cyberarm, I can count on one hand, and while eye implants are a *bit* more common, they're still relatively rare.
So... I made more!

This also spreads out functional cyberlimbs into various techs, as there are now multiple appropriate limbs for each tech tree.

Limbs included:

1. Spacer arm
- Protokinetic Accelerator
  - Can be upgraded
  - Cannot be wielded (but has no accuracy penalty for it)
  - Looses inaccuracy slower than it can fire - pace your shots!
 - Grappling hook
 - Crusher Dagger
2. G.O.R.I.L.L.A arm
 - G.O.R.I.L.L.A gauntlet
 - (very slowly) self-charging C.H.I.M.P
 - Anomaly scanner
3. Angel arm
 - Filtered Autoinjector for Tricordrazine
 - Filtered Autoinjector for Inaprovaline
 - Self-charging defibrillator (3 charges, begins charging 30 seconds after use, about 90 seconds charge time to full charge)
4. Zeus arm
 - Stun fist - one more hit over a stun baton, but harder to aim and slower to charge. Battery-powered.
 - Zeus Fist - punching melee weapon, dealing significant shock damage. Spends a lot of power to EMP on hit - one charge on medium battery.
5. Riplet arm
 - Has one (1) Riplet clamp capable of holding one item or object
6. Persephone arm
 - Mini hoe
 - Clippers
 - Dropper
 - Spade
 - Hatchet
 - Scythe
 - Handheld Grinder
7. Engineer Arm
 - Jaws of Life
 - Power Drill
 - Experimental Welder
 - Standard RCD (*not* experimental)
 - Researchable as a tier 4 industrial tech
 - Costs about as much as a full set of advanced tools + opportunity cost of having to get it installed
8. Hydraulic legs
 - Heavy duty legs able to carry a creature of any size
 - Not slowed down by anything, including carried or worn items (effect diminished when only one leg is installed)
 - Cannot go faster than default speed either
 - Replaces footstep sounds with mech footsteps (Everyone can hear you coming)
9. Pilebunker legs
 - Each installed leg provides one jump with half the distance and cooldown of a Resomi jump
10. Hand implants
- Insulation implant
  - Works like insulated gloves
- Forensics implant
  - Removes your fingerprints, like forensic gloves would
  - Cannot be toggled, so you loose the ability to open mail (which can give them away on antags)
- Claws implant
  - Adds toggle-able claws akin to changeling arm blades
  - Claws can be used like knives
  - Makes a noise when toggled, but is visible only while active
- Thusters implant
  - Wrist-mounted thrusters to keep you mobile in space
11. Diagnostics Implant
- This was pretty much already done, so I finished it while I was at it.
12. Chemistry Implant
- Chemical Analysis Goggles in implant form!
13. Brain implants
- Radio implant emulates a headset with four encryption key slots. Retains encryption keys when implanted or removed
- Expedition radio implant grants access to the expedition comms channel without having to carry a radio. Available at salvage ticket vendor!
- Translator implant, for those foreigner characters who do not want to carry a translator at all times!
14. Nexus Implant
- Comes preinstalled in Avali and Resomi
- Handles the collective mind access to the nexus
- User ID is maintained when implant is transferred to someone else and is copied on cloning (paradox clones retain their ID!) 

Additional changes and fixes I made along the way:
- Cyberarms now come with matching hands, and the hands are what actually deploys the tools. This makes the tool sprite no longer be occluded by the jump suit and the hands, so you may actually get to see them for once. Hair still does override them though.
- Mantis Blades now conform to skin tone, as they were most likely intended to from the start?
- The Mechwright arm is now available as a roundstart trinket for long-time engineer players, analogous to the Ripperdoc arm. It is just a toolbelt, after all, that kind of thing is super easy to get already.
- Reaper arms now have bone gel, completing the surgery kit rather than missing just one tool
- Stun batons can now be powered by powercell. None of the existing batons got cell slots added to them, and they will prefer internal battery storage over an inserted cell
- Hyposprays can now be filtered to only accept specific chems into them
- Cyberlegs can be given a step sound override, applying before Hardsuit sounds
- Cyberlegs can have a minimum and maximum speed modifier, defaulting to x0 and x20
- New EmpOnMeleeHit component
- New LargeGrabber component, allowing an item to act as a hydraulic clamp. Picking up/dropping is toggled via item state and switches automatically when the LargeGrabber is full or empty
- New LimbWithAction component, allowing an action to be attached to an implanted limb
- New InherentAction component doing the same for equipment - this acts exactly like the Jump component, but can be given any action
- New hand implant slots used for hand implants
- New brain implant slot used for brain implants
- Slightly changed surgery steps:
 - Only torso incisions have a ribcage to close. All other bodyparts demand to either mend the skull or restore cartilage instead
 - Rather than adjusting bones, bodyparts other than the torso now reconnect blood vessels instead of prying the bones
 - Slimes no longer have ribcages to close, you instead Restore Slime Mass (using the same tool and time as closing the ribcage)
- Welding implants actually provide welding protection now. (They still provide the 10 second flash eye protection, as before, just also full welding protection, so they work for Resomi and Shadekin rather than doing nothing at all)
- Functional implants actually use the component definitions now, rather than the default component, so sec and syndie implants should work properly as well
- The now functioning syndie eye implant is now available from traitor stores, along with forensics and claw implants
- ReagentGrinderComponent can now be set to not require power, which can be used for lower-end grinders - like a mortar & pestle
- Gibbing a body with implants should no longer crash the server
- Resomi and Avali get access to the Nexus directly through their implant now. This can be reused to make other collective minds via implant, theoretically works in any slot.
- New TaggedOrganComponent. Can add and/or remove tags as desired, and reverts the changes if removed.
- New techs for every discipline to research the new and old cyber arms



TODO:
- do any necessary balance changes
- Probably get my rather shoddy spritework replaced - Maitake did all of the new tools for me already and they look *great*, but I did the claws and limbs myself :sweat_smile: 

Breaks https://github.com/ss14Starlight/space-station-14/pull/2548 , will need to fix either this PR or the other if either gets merged.

## Media (Video/Screenshots)
Implants! (Diagnostics implant already had resources)
<img width="402" height="186" alt="image" src="https://github.com/user-attachments/assets/5bfde80d-719b-4544-8e39-4d32e3d0d68d" />
<img width="434" height="501" alt="grafik" src="https://github.com/user-attachments/assets/f841099e-c9fe-491a-9441-1e45e58efb69" />
In order: Regular Cyberarms, Ripperdoc Arms, Bulwark Arms, Reaper Arms, Mantis Blades (Adjusted to skin tone), Mechwright Arms, Angel Arms, Riplet Arms, Persephone Arms, GORILLA Arms, Spacer Arms, Zeus Arms, Engineer Arms, Hydraulic Legs, Pilebunker Legs
<img width="195" height="271" alt="image" src="https://github.com/user-attachments/assets/776ae0d2-0f47-4a88-aea9-8ed9570242b5" />
<img width="234" height="302" alt="image" src="https://github.com/user-attachments/assets/2b0efad4-9795-4dc0-a0c3-c25ce6d7f60c" />
<img width="212" height="251" alt="image" src="https://github.com/user-attachments/assets/5f8e1886-65ae-438c-976b-9611d30fe451" />
<img width="259" height="304" alt="image" src="https://github.com/user-attachments/assets/305b4a7b-75c2-464f-82b3-00224bbd9196" />
<img width="169" height="212" alt="image" src="https://github.com/user-attachments/assets/3fb13466-84de-4f21-a260-eae34860b995" />
<img width="270" height="294" alt="image" src="https://github.com/user-attachments/assets/cb51fd34-4a18-4ca1-b022-21f904fef50b" />
<img width="264" height="305" alt="image" src="https://github.com/user-attachments/assets/f836bf5b-1f92-4bfd-a74e-886001f00160" />
<img width="175" height="248" alt="image" src="https://github.com/user-attachments/assets/42ceb69b-7c57-4ae3-9be3-8b2257e7624b" />
<img width="213" height="262" alt="image" src="https://github.com/user-attachments/assets/dd7ca973-4a73-41a1-b7d7-eb9e62bac3e0" />
<img width="188" height="250" alt="image" src="https://github.com/user-attachments/assets/2b3bf1f5-57ed-4069-bbe1-caca5485d8f1" />
<img width="139" height="212" alt="image" src="https://github.com/user-attachments/assets/fb7188c5-0133-4aa2-bb5c-1538c5c1b978" />
<img width="158" height="233" alt="image" src="https://github.com/user-attachments/assets/833fb29c-3b61-4e01-876f-c35f7ce32b41" />
<img width="147" height="208" alt="image" src="https://github.com/user-attachments/assets/a682a2e6-a351-433c-b0e8-bdb9502fba4a" />
<img width="155" height="220" alt="image" src="https://github.com/user-attachments/assets/077b5ee3-43a5-4a4d-b906-e84d2df24920" />
<img width="131" height="213" alt="image" src="https://github.com/user-attachments/assets/d675baf6-93a8-4fbb-9dc4-19a4c0961630" />
<img width="118" height="203" alt="image" src="https://github.com/user-attachments/assets/6166675c-748c-46f2-81bd-dc67c61a2c62" />
Claws in both hands
<img width="514" height="659" alt="image" src="https://github.com/user-attachments/assets/e879a496-6cf3-48d3-81e4-dc162e1eff57" />
Claws equipped as gloves
<img width="148" height="240" alt="image" src="https://github.com/user-attachments/assets/ea3bf490-d09d-496f-a497-653065d37bff" />
Cybertools!
<img width="466" height="352" alt="image" src="https://github.com/user-attachments/assets/25cd6bad-e845-4cf9-b7aa-f8092fff0a5f" />

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
:cl: Citrea & sprites by Maitake
- fix: Welding implants work for Shadekin and Resomi now.
- fix: Left cyberlegs no longer move around while in side view.
- tweak: Many surgery steps are now slightly more sensible.
- tweak: Reaper arms now have a bone gel attachment, bringing them up to standard for a full surgical kit.
- tweak: Syndicate eye implants work now and are available from the traitor uplink.
- tweak: Mantis blades now conform to skin tone.
- tweak: Thermal vision cyber eyes are now Arsenal tech
- add: Cyberarms now have corresponding cyberhands, which deploy the tools instead of the arms. Especially noticeable with bulwark arms!
- add: Added Spacer Arm, a cybernetic arm specialised in salvage activities.
- add: Added G.O.R.I.L.L.A Arm, a cybernetic arm specialised in anomalous research.
- add: Added Angel Arm, a cybernetic arm specialised in paramedic duties.
- add: Added Persephone Arm, a cybernetic arm specialised in botany.
- add: Added Zeus Arm, a cybernetic arm sporting a stun fist and melee-range EMP device.
- add: Added Riplet Arm, a cybernetic arm allowing a cargo technician to pick up an entire crate - like a small Ripley!
- add: Added Engineer Arm, a cybernetic arm containing a full advanced tool set and RCD, at a high price.
- add: Added Hydraulic Legs. They are quite loud, but can carry anything you could ever want with ease.
- add: Added Pilebunker Legs to let their users jump.
- add: Added Insulation Hand Implant.
- add: Added Forensics Hand Implant.
- add: Added Claws Hand Implant.
- add: Added Thrusters Hand Implant.
- add: Added Diagnostics Eye Implant - most of this was already done, credit to @Rinary1 
- add: Added Chemistry Eye Implant.
- add: Restored functionality of Syndicate Eye Implant.
- add: Added Radio Brain Implant.
- add: Added Expedition Radio Brain Implant.
- add: Added Translator Brain Implant.
- add: Added Nexus Brain Implant. Not printable, but comes with any Resomi or Avali!
- add: Added Forensics and Claw implants to traitor uplink.
- add: Added Expedition Radio Brain Implant and Spacer Arms to salvage ticket vendor.
- add: Added Mechwright arm to unlockable trinkets (20 hours of Station Engineer)

Citrea-Lingua committed:
Spacer arm

Citrea-Lingua committed:
cyber PKA change

Citrea-Lingua committed:
Hydraulic legs v1

Citrea-Lingua committed:
Sound works now

Citrea-Lingua committed:
Added custom texture layers for new limbs

Citrea-Lingua committed:
Gorilla Arm

Citrea-Lingua committed:
Angel arms

Citrea-Lingua committed:
Shock-hands basics

Citrea-Lingua committed:
removed errant component used for debugging quicker

Citrea-Lingua committed:
Switched shock fist to Zeus Arm

Citrea-Lingua committed:
added stun fist, cyber bone gel and moved surgery arm attachments to cyberlimbs

Citrea-Lingua committed:
Added pilebunker legs

Citrea-Lingua committed:
Added Riplet arm

Citrea-Lingua committed:
Fixed riplet claw and made welding implant actually give welding protection

Citrea-Lingua committed:
Added hand implants and fixed every bodypart having a ribcage

TheRealSlimRedwoods commented:
chooming up with this PR

Citrea-Lingua committed:
Added new limbs to research and brought back unfinished Diagnostics module

Citrea-Lingua committed:
Starlight comments

Citrea-Lingua committed:
Added implants to traitor store

Citrea-Lingua committed:
made syndie eye implants contraband

Citrea-Lingua committed:
turned claws into glove item akin to knuckedusters

Citrea-Lingua committed:
Added Persephone arm

KaiserMaus commented:
What about an auto-implanter for antagonists?
While reading the changes I thought: what if we add eye implants that allow you to see into the distance, like binoculars in-game?

BORTBORTBORTBORTBORT commented:
Please include media of the sprites in game before I can approve them.

Citrea-Lingua commented:
> Please include media of the sprites in game before I can approve them.

Sprites are not finished yet, hence why they are not pushed. Media of them will be included, of course.

Thebigboss1978 commented:
Auto approve

Thebigboss1978 commented:
[
<img width="2048" height="1044" alt="49C45E00-93FB-43AC-90CC-A0C6A888F13B" src="https://github.com/user-attachments/assets/501cb30d-8560-4551-9001-5e9dbcd7420d" />
![IMG_6879](https://github.com/user-attachments/assets/eab70d93-846f-48d0-acae-0a96942064ec)
![3ZSH copy](https://github.com/user-attachments/assets/9d6501d0-b582-4898-9fc7-99cb37e97c9f)
](url)

Citrea-Lingua commented:
Urgh, going to have to look into if this is even still relevant with the changes to medical we may or may not have implemented... When I get around to catching up with what happened 'round here in the past few months (and actually get the sprites done for this :sweat_smile: )

Citrea-Lingua committed:
New sprites for everything (Special thanks to Maitake)

Citrea-Lingua committed:
Fixed left arms putting left hands on your right hand

Citrea-Lingua committed:
Post-merge fixes, including jetpack jumps being broken

Citrea-Lingua committed:
Fixed lathes I broke

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
Had to rebase locally and then it wanted me to manually OK *every* commit since then when syncing - so I force-pushed the rebased branch instead ^^
Making sure now that didn't break anything

Citrea-Lingua committed:
removed extra sprites

Citrea-Lingua commented:
As far as I can tell, I did not break anything by doing that - hooray!
Unless I missed something :sweat_smile:

Citrea-Lingua commented:
I *did*, however, break my changes to the jump system in the process, so let me fix that.

Citrea-Lingua committed:
reverted changes to sharedjumpsystem, as they are no longer needed

Citrea-Lingua commented:
Looks like Rinary1 already fixed the jump system, so my changes are not needed anyway - reverted.

Citrea-Lingua committed:
Adjusted leg animation timers

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
Received a go-ahead from Maitake; updating the meta.json files to the new format and then I will be undrafting!

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
moved new attributions to the new format

Citrea-Lingua committed:
reverted mistakenly left in identifier of which tools were taken from tgstation (not necessary after moving that attribution to the concrete sprites in question)

Citrea-Lingua committed:
fixed runner verification

Citrea-Lingua commented:
This will likely need balance changes.

Citrea-Lingua committed:
re-adding moved copyright key again
It has to be file-wide in addition to the individual sprite attributions, it seems, even if they say the same thing

Citrea-Lingua committed:
Adjusted sprite and text in uplink catalogue

Citrea-Lingua committed:
missing S from guidebook entry tag

Citrea-Lingua committed:
Tiny power cells do not have a charge indicator

Citrea-Lingua committed:
added a default to fix phantom blade spell

Citrea-Lingua committed:
Cyberdropper was expecting to have inhand fill states despite never needing them.

Citrea-Lingua committed:
Added brain implants!

Citrea-Lingua commented:
working on brain implants and a chem goggle implant now, if this somehow gets merged before those are done I will do them as a new PR :sweat_smile:

Citrea-Lingua committed:
Translator implant now updates properly

Citrea-Lingua committed:
New implant textures and chemistry eye implant

Citrea-Lingua committed:
Nexus implant now retains collective mind identity

Citrea-Lingua committed:
Nexus ID now gets copied on cloning

Citrea-Lingua committed:
Radio Implant!

Citrea-Lingua committed:
Added implant recipes

Citrea-Lingua committed:
Radio implant works now and gibbing a character with implants installed no longer crashes the game

Citrea-Lingua committed:
Radio implant and expedition radio implant split, radio implant can now accept encryption keys

Citrea-Lingua committed:
added description to spacer arms

Citrea-Lingua committed:
collective mind data saved in implants not gets overwritten if data was already present in collective mind

Citrea-Lingua committed:
Revert "collective mind data saved in implants not gets overwritten if data was already present in collective mind"

This reverts commit ec675aed7860cc5a44ffc0f2b1abfaab2e93304a.
Whilte technically creating edge cases, reverting this should be slightly more performant, and there is no reasonable scenario where those edge cases *could* even appear.

Citrea-Lingua committed:
adjusted nexus implant name and description

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
Guh, no idea whay these tests are failing; investigating

Citrea-Lingua committed:
Hydraulic legs are now slower by default

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Happyrobot33 commented:
@Citrea-Lingua this close to done ish?

Citrea-Lingua committed:
post merge fixes

Citrea-Lingua commented on Content.Server/Stunnable/Systems/StunbatonSystem.cs:
```diff
@@ -33,11 +35,17 @@ public override void Initialize()
-            !TryComp<PredictedBatteryComponent>(entity.Owner, out var battery) || !_battery.TryUseCharge((entity.Owner, battery), entity.Comp.EnergyPerUse))
+            !(TryComp(entity.Owner, out PredictedBatteryComponent? battery) ||
+            _powerCell.TryGetBatteryFromSlot(entity.Owner, out batteryEntity)) ||
+            !_battery.TryUseCharge(batteryEntity.HasValue ? batteryEntity.Value.AsNullable() : (entity.Owner, battery), entity.Comp.EnergyPerUse))
```
This used to be so elegant, and then things got moved to prediction and entities and now it's clunky and bad :c

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
> @Citrea-Lingua this close to done ish?

Working on fixing it after the upstream merge. Once it works again, it should be done.

Citrea-Lingua committed:
useRate -> useCharge

Citrea-Lingua commented:
Right. This test error again. Which I don't have anything to do with, so far as I can tell - but kind of *must* have caused..?

Citrea-Lingua committed:
Added missing check for component before using it

Citrea-Lingua committed:
added containercontainer to radio brain implant

Citrea-Lingua commented:
It is now done, barring any balance concerns! ^^

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Conflee reviewed (approved):
looks good

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

TacoLlama commented:
this PR is so cool, also very excited tor the LargeGrabber comp, been wanting something like that.

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Huaqas commented:
Cybernetics and Implants could really use a Guidebook page explaining what they do.

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
Not wrong, that should perhaps be a separate PR soon. Putting it on my to do list after borg hats, but whether I actually get around to it... :sweat_smile:

redboit commented:
yay!

redboit commented:
im stupid and cant read, does this allow for more roundstart cyberlimbs? like the ones in the cybernetics menu?

Citrea-Lingua commented:
No, this is expressly something to be researched and then you get it implanted by medical - maybe surgeons will finally get some work every shift, rather than just occasionally :sweat_smile:

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
Been keeping my PRs up to date, since it looks like stuff is being merged again - this one seems to be the only one having to deal with merge conflicts. Serves me right for letting scope creep get the better of me :sweat_smile:

Citrea-Lingua commented:
This test failure unfortunately looks genuine, though I have no idea how. Investigating.

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
Oooor not?

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Thebigboss1978 commented:
can we live now = please apply the changes

Thebigboss1978 commented:
merge

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua commented:
As far as I can tell, the test fails here are invalid, even if they are weirdly consistently flakey. May want a second pair of eyes on them.

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Happyrobot33 reviewed (changes requested):
needs code changes sorrounded by starlight start starlight end comments

Citrea-Lingua commented:
yea, looks like a few spots got missed.
*sigh*
Alright, going through and checking :sweat_smile:

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Starlight comment & reverted accidental reaper arm nerf

Citrea-Lingua commented:
All starlight changes outside of _Starlight files should now be marked. And I had accidentally nerfed the reaper arm during a merge conflict resolution - fixed that :sweat_smile:

Citrea-Lingua committed:
flipped hydraulic leg animation frames

Citrea-Lingua commented:
Hydraulic legs had the animation frames in the wrong order, fixed now. Not sure if @Conflee wants to re-review them - but just in case: it is *specifically* the left and right hydraulic legs and there were 0 changes to the actual sprites, just the order of the frames changed :sweat_smile:

Citrea-Lingua commented:
This one's a flakey test, far as I can tell.

Citrea-Lingua commented:
Expanding on this to fix the weirdness with cyberhands and them blocking the tool graphics.

Citrea-Lingua committed:
Cyberhands do the blinking for the arms now

Citrea-Lingua committed:
Cyberarms now have corresponding cyberhands

Citrea-Lingua commented:
Lots of sprite changes, going to need another sprite review from @Conflee (sorry :sweat_smile: )

Making screenshots, they will be updated momentarily.

Citrea-Lingua committed:
fixed meta file

Citrea-Lingua committed:
Cyberlimbs now actually add their children instead of just their visuals

Citrea-Lingua committed:
Fixed mantis blades and extra hands

Citrea-Lingua committed:
Added engineer arm

Citrea-Lingua commented:
Added an additional arm, made changes to how they are researched, and added the Mechwright arm as an unlockable trinket analogous to the Ripperdoc arm.

Citrea-Lingua committed:
yml fix

Citrea-Lingua committed:
renamed functional cyberlimbs tech

Citrea-Lingua committed:
Tiny sprite adjustment

Citrea-Lingua committed:
Fix actions not getting removed properly

Citrea-Lingua commented:
SOmething is messy with how the hands are attached now... Working on it.

Citrea-Lingua committed:
Hands (and feet) should now be properly removed?

Citrea-Lingua commented:
Going to do more work on this tomorrow. Sprites are ready to be reviewed, the code functionality is not fully there yet.
Hands are being *really* weird.

Citrea-Lingua committed:
reverted trying to fix this mess, trying again from scratch

Citrea-Lingua committed:
adding hands adds children

Citrea-Lingua committed:
When removing a limb with item, the body is now the performer as it should be

Citrea-Lingua committed:
A little cleanup

Citrea-Lingua commented:
Alright, it is ready to go, though I am still investigating a bug where the hand graphics do not update the UI until it is reloaded, adding the button *in the wrong location* again. This is a bug in dev though, and not linked to my changes.

Citrea-Lingua committed:
fixed discipline

Citrea-Lingua committed:
Changed mantis blade sprites

Citrea-Lingua committed:
Mantis blade no longer inherits colour when deployed

Citrea-Lingua committed:
Both states need to nor match skin to work, apparently

Citrea-Lingua commented:
Resomiparts to match are now also done. (not yet part of this PR, will be added in or after https://github.com/ss14Starlight/space-station-14/pull/2548 )

redboit commented:
wait, is the cell slot for the holdable batons or the implant one?

Citrea-Lingua commented:
the cyberarm baton has a cell slot, *but* any baton could be *given* one and draw from a power cell, if that was desired. I did not add a power cell slot to any existing batons.

redboit commented:
> the cyberarm baton has a cell slot, _but_ any baton could be _given_ one and draw from a power cell, if that was desired. I did not add a power cell slot to any existing batons.

okay thx ily

Citrea-Lingua commented:
Test fail looks invalid.

Conflee reviewed (approved):
sprites good

walksanatora commented on Content.Server/_Starlight/Emp/EmpOnMeleeHitComponent.cs:
```diff
@@ -0,0 +1,32 @@
+[Access(typeof(Content.Server.Emp.EmpSystem))]
+public sealed partial class EmpOnMeleeHitComponent : Component
+{
+    [DataField("range"), ViewVariables(VVAccess.ReadWrite)]
```
you dont need to specify name when it is the same but camel case. eg: `Range` becomes `range` in yaml automatically. `EnergyConsumption` -> `energyConsumption`. and DataField implies VV + VVAccess.ReadWrite so that is also redundant.

Citrea-Lingua replied:
I am aware, why did I do it like that..? :sweat_smile:
Changed

walksanatora commented on Content.Server/_Starlight/Equipment/Components/LargeGrabberComponent.cs:
```diff
@@ -0,0 +1,54 @@
+    /// The change in energy after each grab.
+    /// </summary>
+    [DataField("grabEnergyCost")]
+    public float GrabEnergyCost = 30;
```
explicit name matches auto generated name.

Citrea-Lingua replied:
fixed

walksanatora commented on Content.Server/Emp/EmpSystem.cs:
```diff
@@ -2,14 +2,32 @@
+    // 🌟Starlight🌟 end
+
+    public const string EmpPulseEffectPrototype = "EffectEmpPulse";
+
```
Starlight End should be down here

Citrea-Lingua replied:
Looks like that one slipped in during a merge - that constant was moved upstream, hence why it's not marked. Removed, as it's no longer supposed to be there and had 0 references.

walksanatora commented on Resources/Prototypes/_StarLight/Body/thaven.yml:
```diff
@@ -12,6 +12,7 @@
         eyes: OrganHumanEyes
         tongue: OrganHumanTongue
         eye_implant: null
+        brain_implant: null
```
thaven need hand implant slots also

Citrea-Lingua replied:
Oop, you are correct.

Citrea-Lingua committed:
review fixes

Citrea-Lingua committed:
Another reviewed change

Citrea-Lingua commented:
All checks green, ready for re-review ^^

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

Citrea-Lingua committed:
Merge branch 'starlight-dev' into More-Cyberlimbs

walksanatora reviewed (approved):
Todays your BIG DAY!!!
